go/vt/vtgate/planbuilder/testdata/unsupported_cases.json

```diff
@@ -383,5 +373,10 @@
     "comment": "select (select 1 from user u having count(ue.col) > 10) from user_extra ue",
     "query": "select (select 1 from user u having count(ue.col) > 10) from user_extra ue",
     "plan": "VT12001: unsupported: correlated subquery is only supported for EXISTS"
+  },
+  {
+    "comment": "CTEs cant use a table with the same name as the CTE alias",
+    "query": "with user as (select aa from user where user.id=1) select ref.col from ref join user",
+    "plan": "VT12001: unsupported: do not support CTE that use the CTE alias inside the CTE query"
   }
```

Comment:
yeah, long term I think we want much better support for CTEs. instead of rewriting them to derived tables they should be first hand citizens, so that the parts of a query that we push down to MySQL uses CTE if the original query used a CTE, and even supporting recursive CTEs using a new engine primitive

Reply:
that makes sense!